### PR TITLE
front: fix "invalid Date" errors

### LINKED ISSUE #7836: "Invalid date" displayed in operational studies
### What happened?

When creating a project / study / scenario the front displays "updated on invalid date" / "modifiée le invalid date". 

See:
image

### What did you expect to happen?

I would expect it to display a normal date, or no date at all

### How can we reproduce it (as minimally and precisely as possible)?

1. Create any project, study or scenario ?

### What operating system, browser and environment are you using?

- **Browser**: Firefox
- **OS**: Windows 11
- **Env**: osrd.dev.dgexsol.fr

### OSRD version (top right corner `Account` button > `Informations`)

24485e2

### PR DESCRIPTION
**Pull request by emersion**
_See individual commits._

Closes: https://github.com/OpenRailAssociation/osrd/issues/7836

### CODE REVIEW

**Comment by theocrsb** [comment]
LGTM

File: front/src/utils/date.ts [hunk 1]
```diff
@@ -19,16 +18,19 @@ export function formatIsoDate(date: Date) {
...
-      dayjs.locale('fr');
+      locale = 'fr';
       break;
     default:
-      dayjs.locale('en-gb');
+      locale = 'en';
   }
-  const dateToUTC = dayjs(`${date}Z`); // The 'Z' is to ensure we have an UTC date
+  // Force interpreting the date in UTC
+  const dateToUTC = dayjs(date).utc(true);
```
Comment (emersion) [inline front/src/utils/date.ts:30]: Maybe we should just drop the UTC conversion altogether here? (In practice browsers ignore the `Z` suffix, so dropping the UTC conversion would retain current behavior. I tried to keep the UTC conversion to retain the original intent of the old code, but maybe that's not a good idea.)

**Review by Math-R — APPROVED** [review]
tested and seems fine to me

File: front/src/utils/date.ts (hunk 1 above)
Comment (Math-R) [reply front/src/utils/date.ts:30]: we can discuss that if you want but it's seems good to me

Comment (emersion) [reply front/src/utils/date.ts:30]: Alright, let's leave it as-is for now. Would be nice to audit all of the call-sites at some point to find out if any `Date` values really need this.

**Review by theocrsb — APPROVED** [review]

**Pull request merged**